src/Hook.lua

```diff
@@ -97,7 +103,9 @@ end
 -- @return the results of the hook if it exists.
 function M.apply(name, ...)
    if (validT[name]) then
-      return validT[name](...)
+      for i=1,#validT[name] do
+         validT[name][i](...)
```

Comment:
NB, I don't return anything anymore, as it is not clear what should be returned. I considered a table of return values

```
      local returnT = {}
      for i=1,#validT[name] do
         local return_val = validT[name][i](...)
         if return_val ~= nil then
            table.insert(returnT, return_val)
         end
      end
      return returnT
```
but the awkward part is that tables cant contain nil values, so the returning table could have less elements than the original number of functions in the hook table. That makes it impossible to do reasonable handling of return code (which is pretty hard anyway since the hooks are arbitrary functions being executed, so you have no idea what different return values mean in any case...).